executor/include/execution/cuda_executor.cu

```diff
@@ -0,0 +1,6 @@
+#include "cuda_executor.cuh"
+
+template <typename Function>
+__global__ void global_kernel(Function f) {
+  f();
+}
```

Comment:
Template definitions need to be shipped just like headers. Putting this inside it's own .cu file is a bit misleading.

Reply:
Resolved in #12